kairos-deposit-contract/deposit-session/src/main.rs

```diff
@@ -0,0 +1,24 @@
+#![no_std]
+#![no_main]
+use casper_contract::contract_api::{account, runtime, system};
+use casper_types::{runtime_args, AddressableEntityHash, URef, U512};
+
+#[no_mangle]
+pub extern "C" fn call() {
+    let contract_hash: AddressableEntityHash = runtime::get_named_arg("deposit_contract");
+    let amount: U512 = runtime::get_named_arg("amount");
+    let source: URef = account::get_main_purse();
+    // create a temporary purse that can be passed to the deposit contract
```

Comment:
It would be good if you can explain what purpose the temporary purse serves. Do we need an intermediate purse to deposit to our contract's purse? is it then `owner_purse` -> `tem_purse` -> `contract_purse`

Reply:
Please document this here such that someone revisiting this code is aware